content/docs/formal-verification/tamarin/00-basic-concepts.md

```diff
@@ -0,0 +1,281 @@
+brew install tamarin-prover/tap/tamarin-prover
+```
+
+{{< /tab >}}
+
+{{< tab "Using Pacman" >}}
+On Arch Linux you can install Tamarin using Pacman with the following command:
+
+```sh
+pacman -S tamarin-prover
+```
+
+{{< /tab >}}
+
+{{< tab "Using the Nix package manager" >}}
+You can install Tamarin from Nixpkgs using the Nix package manager with the following command:
+
+```sh
+nix-env -i tamarin-prover
+```
+
+{{< /tab >}}
+
+{{< tab "On NixOS" >}}
+If you are running NixOS you can install Tamarin by adding `tamarin-prover` to your `environment.systemPackages`.
+
+{{< /tab >}}
+
+{{< /tabs >}}
+
+## Tamarin basic concepts
+
+Tamarin models can be specified either using [multiset](https://en.wikipedia.org/wiki/Multiset) rewrite rules, or using
+a process calculus based on a variant of the applied pi-calculus. The global protocol state is given by a multiset of
+facts that is expanded and updated as the protocol progresses. Here, a fact is essentially a statement about a set of
+terms. For example, _`Alice` has registered the public key `pk` with the server_, or _The attacker has learned the
+private key `sk`_. Protocol properties are expressed as properties of the set of possible executions of the protocol,
+and are specified using a fragment of temporal logic. (This means that we can express properties of terms at different
+points in time.)
+
+### Variables and sorts
+
+Variables in Tamarin range over five different sorts (or types). There is a top sort for messages that can be sent over
+the network, with subsorts for public values, fresh (or private) values, and natural numbers. There is also a sort for
+temporal values (essentially, points in time). To make the sort clear from context, each sort is prefixed by a single
+character.
+
+- `~x` denotes a fresh value. _Fresh values_ are random values like secret keys
+  and nonces and are not known to the attacker.
+- `$x` denotes a public value. _Public values_ are publicly known. In
+  particular, they are known to the attacker. Constant public values are
+  written as strings (without the `$`). For example, the constant 0x30 is
+  written `'0x30'`, and a constant generator _g_ of a cyclic group is written
+  `'g'`. (For example, the public key corresponding to the private key `~x` in
+  a Diffie-Hellman based protocol is written `'g'^~x`.)
+- `%n` denotes a natural number. _Natural numbers_ start at `%1` in Tamarin and
+  require the `natural-numbers` builtin to use. They are used to represent
+  (small) numbers like counters, and are assumed to be known to the attacker.
+- `#i` denotes a temporal value. _Temporal values_ are not used to describe the
+  protocol directly, but are used to express protocol properties like "If the
+  attacker knows Alice's key at time `#j`, then Alice's key must have been
+  leaked at some time `#i` before `#j`."
+
+Terms may be composed using functions. These are either user defined, or defined by importing a predefined theory (known as a `builtin`).
+
```

Comment:
Can we get a very simple example before we continue? I think readers would like to know what they're getting into =)